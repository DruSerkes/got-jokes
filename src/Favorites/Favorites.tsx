import * as React from 'react';
import { jokeData } from '../types';
import { Favorite } from '../Favorite/Favorite';

interface FavoritesProps {
    favorites?: jokeData[];
    clearFavorites: (id: Number) => void;
}

export const Favorites = ({ favorites, clearFavorites }: FavoritesProps) => {
    return (
        <>
            <h3>Favorites</h3>
            <ul className='Favorites rounded list-group my-3'>
                {favorites?.map(fav => <Favorite key={fav.id} joke={fav} clearFavorites={clearFavorites} />)}
            </ul>
        </>
    )
}