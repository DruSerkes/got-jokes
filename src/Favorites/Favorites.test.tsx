import * as React from 'react';
import { fireEvent, render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import { Favorites } from './Favorites';



describe('Favorites tests', () => {
    
})