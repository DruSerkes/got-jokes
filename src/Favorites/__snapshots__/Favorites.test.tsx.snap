// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites tests should render and match snapshot 1`] = `
<DocumentFragment>
  <h3>
    Favorites
  </h3>
  <ul
    class="Favorites rounded list-group my-3"
  >
    <li
      class="py-3 list-group-item-dark border-bottom rounded border-dark"
    >
      test joke
      <button
        class="Favorite-removeButton btn"
        type="button"
      >
        X
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
